### ISSUE TITLE
Documentation request: Annotator use case

### ISSUE THREAD

**Issue by iainmwallace** [solvebio/solvebio-r#33]
Hi,

Annotator function is awesome! I would like to use it to annotate an ad hoc dataset that has a primary id via a shiny app. 

Can you provide some R examples of how to
- list all datasets my primary id list appears in?
   (ideally limited to a specific vault, and a count of id appearances from list in each dataset)
- list all fields of these datasets
- efficient method of adding fields to my original dataset 
      (it isn't clear if I should annotate all at once or create intermediate datasets that I delete)
- delete a dataset that I have created

Thanks

Iain

**Comment by davecap** [comment]
Thanks for the feedback, glad you like the annotator! In short, you should use the Annotator to transform records that are stored locally in your R session, and use "Dataset Migrations" to transform datasets that are already on SolveBio.

In the end, it is up to you how you want to process the data. In many cases you can do a bunch of annotations in one step, but you can use intermediate datasets for transforms that require more than one step (such as merging/comparing datasets).

I'll together some examples for you and also add them to the developer docs.

**Comment by iainmwallace** [comment]
Thanks! Can you add an example of how to follow the progress of a migration? It doesn't seem to appear in the activity tab until it is complete and present in the activity tab of the current UI

**Comment by davecap** [comment]
Good point, this will be fixed in the new version of the site. For now, you can use some R code to follow along:

```r
migration = DatasetMigration.create(...)

while(migration$status == "queued" || migration$status == "running") {
    migration <- DatasetMigration.retrieve(migration$id)
    cat(migration$status, sep="\n")
    Sys.sleep(3)
}
```

**Comment by iainmwallace** [comment]
Thanks!
